script.js

```diff
@@ -0,0 +1,155 @@
+  if (!selectedOption) {
+    alert('Please select an answer.');
+    return;
+  }
+
+  const currentQuestion = quizQuestions[currentQuestionIndex];
+  const isCorrect = currentQuestion.options.find(
+    option => option.text === selectedOption.value
+  ).isCorrect;
+
+  if (isCorrect) {
+    alert(`Correct answer. Explanation: ${currentQuestion.explanation}`);
+    if (currentPlayer === 1) {
+      player1Score++;
+      player1ScoreHtml.textContent = player1Score;
+    } else {
+      player2Score++;
+      player2ScoreHtml.textContent = player2Score;
+    }
+  } else {
+    alert(`Wrong answer. Explanation: ${currentQuestion.explanation}`);
+  }
+
+  currentQuestionIndex++;
+  if (currentQuestionIndex < quizQuestions.length) {
+    displayQuestion();
+    if (currentPlayer === 1) {
+      currentPlayer = 2;
+    } else {
+      currentPlayer = 1;
+    }
+    currentTurnHtml.textContent = `Player ${currentPlayer}`;
+  } else {
+    alert('quiz completed!');
+    form.removeEventListener('submit', submitForm);
+    form.innerHTML = '';
+    showQuestions(); 
+    stateWinner();
+  }
+}
+
+function showQuestions() {
+  questionList.innerHTML = ''; 
+  questionList.style.display = "flex"
+  questionList.style.height = "400px"
+
+  quizQuestions.forEach((question, index) => {
+    const questionDiv = document.createElement('div');
+    questionDiv.className = 'question-div';
+    questionDiv.innerHTML = `
+      <h3>${index + 1}. ${question.question}</h3>
+      <ul>
+        ${question.options.map(option => `<li>${option.text}</li>`).join('')}
+      </ul>
+      <button class="reveal-button"onclick="revealAnswer(${index})">Reveal Answer</button>
+      <p id="answer-${index}" style="display: none;">Correct Answer: ${
+      question.options.find(option => option.isCorrect).text
+    }</p>
+    `;
+    questionList.appendChild(questionDiv);
+  });
+
+  
+}
+
+function revealAnswer(index) {
+  document.getElementById(`answer-${index}`).style.display = 'block';
+}
+
+function stateWinner() {
+  const winningMessage = document.createElement('p');
+  const player1ValueName = document.getElementById('player1ValueName').value;
+  const player2ValueName = document.getElementById('player2ValueName').value;  
+
+  if (player1Score > player2Score) {
+    winningMessage.textContent = `${player1ValueName} Won`;
+  } else if (player2Score > player1Score) {
+    winningMessage.textContent = `${player2ValueName} Won`;
+  } else {
+    winningMessage.textContent = "Its a tie!";
+  }
+  document.body.appendChild(winningMessage);
+}
+
+function filterQuestions(searchTerm) {
+  questionList.innerHTML = '';
+
+  quizQuestions.forEach(question => {
+    if (question.question.toLowerCase().includes(searchTerm.toLowerCase())) {
+      const questionDiv = document.createElement('div');
+      questionDiv.textContent = question.question;
+      questionList.appendChild(questionDiv);
+    }
+  });
+}
```

Comment:
I see that you have implemented the searching functionality, but the assignment also asked to have the option of random sorting or alphabetical sorting of questions. I do not see that being implemented.